modules/core/src/main/java/org/apache/synapse/util/InlineExpressionUtil.java

```diff
@@ -197,4 +202,48 @@ private static boolean isValidXML(String stringToValidate) {
         }
         return false;
     }
+
+    /**
+     * Checks whether inline text contains synapse expressions
+     * Inline expressions will be denoted inside ${}
+     * e.g.: ${var.var1}, ${payload.element.id}
+     *
+     * @param inlineText Inline text string
+     * @return true if the string contains inline synapse expressions, false otherwise
+     */
+    public static boolean isInlineSynapseExpressionsContentAware(String inlineText) {
+
+        Matcher matcher = SYNAPSE_EXPRESSION_PLACEHOLDER_PATTERN.matcher(inlineText);
+        while (matcher.find()) {
+            // Extract the expression inside ${...}
+            String placeholder = matcher.group(1);
+            if (placeholder.contains("xpath(") || placeholder.contains("payload.") || placeholder.contains("$.")) {
```

Comment:
we might have a prob here for instances like ${toUpper(payload.name)}

Reply:
I used the same content aware check we have in the Synapse expressions here as well.

Reply:
+1